accounting/gst.js

```diff
@@ -252,3 +252,85 @@ async function generateB2csData(invoices) {
 
   return b2cs;
 }
+
+export async function generateGstr2Csv(getReportData) {
+  const { gstin } = frappe.AccountingSettings;
+  if (!gstin) {
+    showMessageDialog({
+      message: _('Export Failed'),
+      description: _('Please set GSTIN in General Settings.'),
+    });
+    return;
+  }
+
+  const {
+    rows,
+    columns,
+    filters: { transferType, toDate },
+  } = getReportData();
+
+  const { filePath, canceled } = await getSavePath('gstr-2', 'csv');
+  if (canceled || !filePath) return;
+
+  let gstData;
+  if (transferType === 'B2B') {
+    gstData = await generateB2bCsvGstr2(rows, columns);
+  }
```

Comment:
There's only one transfer type so is this needed? Why have we removed other transfer types @piyushsinghania ?

Are they not present in EN also?

Reply:
Yes other transfer types like B2CS, B2CL are not there in EN. Since GSTR-2 deals with purchase invoice, gst holders can only give proper gst invoices i.e. B2B.

Kept the if check there because, in EN there is another transfer type, something called CNDN I guess, related to credit debit. So, in future if we are going to add that option then it would make sense.